libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/SmartSqlHelper.java

```diff
@@ -120,16 +120,24 @@ else if (path.equals(SmartStore.SOUP_LAST_MODIFIED_DATE)) {
 				// {soupName:path}
 				else {
 					String columnName = getColumnNameForPathForSmartSql(db, soupName, path, position);
-					matcher.appendReplacement(sql, columnName);
+					matcher.appendReplacement(sql, columnName.replace("$", "\\$") /* treat any $ as litteral */);
```

Comment:
When doing select {soup:path} on a json1 indexed path, you will get sql that looks like:
select json_extract(soup, '$.x.y.z')
We don't want appendReplacement to think $ is referring to a previous match